cpp/src/io/orc/writer_impl.cu

```diff
@@ -1234,19 +1243,20 @@ void writer::impl::write(table_view const &table)
     ff.types[0].subtypes.resize(num_columns);
     ff.types[0].fieldNames.resize(num_columns);
     for (auto const &column : orc_columns) {
-      ff.types[1 + column.id()].kind      = column.orc_kind();
-      ff.types[0].subtypes[column.id()]   = 1 + column.id();
-      ff.types[0].fieldNames[column.id()] = column.orc_name();
+      ff.types[column.id()].kind             = column.orc_kind();
+      ff.types[0].subtypes[column.index()]   = column.id();
```

Comment:
This change looks confusing, because `id()` has been renamed to `index()`, and `id()` has then been added (returns index+1).